src/core/url.ts

```diff
@@ -0,0 +1,74 @@
+export type Locatable = URL | string
+
+export function expandPath(pathOrUrl: string | URL): URL {
+  const anchor = document.createElement("a") as HTMLAnchorElement
+  anchor.href = pathOrUrl.toString()
+
+  return new URL(anchor.href)
+}
+
+export function anchor(url: URL): string {
```

Comment:
@domchristie I've done my best to transform your version to one that is a function and not an instance method.

Reply:
Nice! I think I got distracted by lines 18-23, and missed 11-16 👍

Reply:
Oh no! That's git debris I need to remove. Thanks for the heads up.